### PR TITLE
feat(services): provide a linstor-monitor daemon to check master changes

### PR DESCRIPTION
**Pull request by Wescoeur**

### CODE REVIEW

File: linstor/linstor-monitord.c [hunk 1]
```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (C) 2019  Vates SAS - ronan.abhamon@vates.fr
```
Comment (olivierlambert) [inline linstor/linstor-monitord.c]: 2020

**Review by olivierlambert — CHANGES_REQUESTED** [review]

File: linstor/linstor-monitord.c [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+#define POOL_CONF_FILE "pool.conf"
+#define POOL_CONF_ABS_FILE POOL_CONF_DIR "/" POOL_CONF_FILE
+
+// In milliseconds.
+#define POLL_TIMEOUT 2000
+
+// -----------------------------------------------------------------------------
+
+static inline void normalizeTime (struct timespec *spec) {
+  while (spec->tv_nsec >= 1000000000) {
```
Comment (benjamreis) [inline linstor/linstor-monitord.c:47]: Why not put `1000000000` in a variable?
It's used 3 times and it'd be easier to read.

File: linstor/linstor-monitord.c [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+  while (spec->tv_nsec < 0) {
+    --spec->tv_sec;
+    spec->tv_nsec += 1000000000;
+  }
+}
+
+static inline struct timespec getCurrentTime () {
+  struct timespec spec;
+  clock_gettime(CLOCK_MONOTONIC, &spec);
+  return (struct timespec){
```
Comment (benjamreis) [inline linstor/linstor-monitord.c:60]:
```suggestion
  return (struct timespec) {
```

File: linstor/linstor-monitord.c [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+  // Main process.
+  int status;
+  if (waitpid(pid, &status, 0) < 0) {
+    syslog(LOG_ERR, "Failed to wait command: `%s`.", *argv);
+    return -errno;
+  }
+
+  if (WIFEXITED(status)) {
+    const int code = WEXITSTATUS(status);
+    if (WEXITSTATUS(status) == 0)
```
Comment (benjamreis) [inline linstor/linstor-monitord.c]: Why not use `code`?
```suggestion
    if (code == 0)
```

File: linstor/linstor-monitord.c (hunk 3 above)
Comment (Wescoeur) [reply linstor/linstor-monitord.c:60]: It's a cast. So no space seems good. :)

File: linstor/linstor-monitord.c (hunk 4 above)
Comment (Wescoeur) [reply linstor/linstor-monitord.c]: :+1:

File: linstor/linstor-monitord.c (hunk 2 above)
Comment (Wescoeur) [reply linstor/linstor-monitord.c:47]: Well I don't know if it would be easier to read, but if you have an explicit variable name, don't hesitate. :p

**Pull request merged**